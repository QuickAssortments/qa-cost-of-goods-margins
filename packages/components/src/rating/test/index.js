/** @format */
/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import Rating from '../';
import ProductRating from '../product';
import ReviewRating from '../review';

describe( 'Rating', () => {
	test( 'should render the passed rating prop', () => {
		const rating = shallow( <Rating rating={ 4 } /> );
		expect( rating ).toMatchSnapshot();
	} );

	test( 'should render the correct amount of total stars', () => {
		const rating = shallow( <Rating rating={ 4 } totalStars={ 6 } /> );
		expect( rating ).toMatchSnapshot();
	} );

	test( 'should render stars at a different size', () => {
		const rating = shallow( <Rating rating={ 1 } size={ 36 } /> );
		expect( rating ).toMatchSnapshot();
	} );
} );

describe( 'ReviewRating', () => {
	test( 'should render rating based on review object', () => {
		const review = {
			review: 'Nice T-shirt!',
			rating: 1.5,
		};
		const rating = shallow( <ReviewRating review={ review } /> );
		expect( rating ).toMatchSnapshot();
	} );
} );

describe( 'ProductRating', () => {
	test( 'should render rating based on product object', () => {
		const product = {
			name: 'Test Product',
			average_rating: 2.5,
		};
		const rating = shallow( <ProductRating product={ product } /> );
		expect( rating ).toMatchSnapshot();
	} );
} );
