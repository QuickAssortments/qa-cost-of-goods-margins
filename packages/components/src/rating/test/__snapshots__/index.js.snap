// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRating should render rating based on product object 1`] = `
<Rating
  rating={2.5}
  size={18}
  totalStars={5}
/>
`;

exports[`Rating should render stars at a different size 1`] = `
<div
  aria-label="1 out of 5 stars."
  className="woocommerce-rating"
>
  <t
    icon="star"
    key="star-0"
    size={24}
    style={
      Object {
        "height": "36px",
        "width": "36px",
      }
    }
  />
  <t
    icon="star"
    key="star-1"
    size={24}
    style={
      Object {
        "height": "36px",
        "width": "36px",
      }
    }
  />
  <t
    icon="star"
    key="star-2"
    size={24}
    style={
      Object {
        "height": "36px",
        "width": "36px",
      }
    }
  />
  <t
    icon="star"
    key="star-3"
    size={24}
    style={
      Object {
        "height": "36px",
        "width": "36px",
      }
    }
  />
  <t
    icon="star"
    key="star-4"
    size={24}
    style={
      Object {
        "height": "36px",
        "width": "36px",
      }
    }
  />
  <div
    className="woocommerce-rating__star-outline"
    style={
      Object {
        "width": "20%",
      }
    }
  >
    <t
      icon="star"
      key="star-0"
      size={24}
      style={
        Object {
          "height": "36px",
          "width": "36px",
        }
      }
    />
    <t
      icon="star"
      key="star-1"
      size={24}
      style={
        Object {
          "height": "36px",
          "width": "36px",
        }
      }
    />
    <t
      icon="star"
      key="star-2"
      size={24}
      style={
        Object {
          "height": "36px",
          "width": "36px",
        }
      }
    />
    <t
      icon="star"
      key="star-3"
      size={24}
      style={
        Object {
          "height": "36px",
          "width": "36px",
        }
      }
    />
    <t
      icon="star"
      key="star-4"
      size={24}
      style={
        Object {
          "height": "36px",
          "width": "36px",
        }
      }
    />
  </div>
</div>
`;

exports[`Rating should render the correct amount of total stars 1`] = `
<div
  aria-label="4 out of 6 stars."
  className="woocommerce-rating"
>
  <t
    icon="star"
    key="star-0"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-1"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-2"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-3"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-4"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-5"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <div
    className="woocommerce-rating__star-outline"
    style={
      Object {
        "width": "67%",
      }
    }
  >
    <t
      icon="star"
      key="star-0"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-1"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-2"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-3"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-4"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-5"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
  </div>
</div>
`;

exports[`Rating should render the passed rating prop 1`] = `
<div
  aria-label="4 out of 5 stars."
  className="woocommerce-rating"
>
  <t
    icon="star"
    key="star-0"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-1"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-2"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-3"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <t
    icon="star"
    key="star-4"
    size={24}
    style={
      Object {
        "height": "18px",
        "width": "18px",
      }
    }
  />
  <div
    className="woocommerce-rating__star-outline"
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <t
      icon="star"
      key="star-0"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-1"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-2"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-3"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
    <t
      icon="star"
      key="star-4"
      size={24}
      style={
        Object {
          "height": "18px",
          "width": "18px",
        }
      }
    />
  </div>
</div>
`;

exports[`ReviewRating should render rating based on review object 1`] = `
<Rating
  rating={1.5}
  size={18}
  totalStars={5}
/>
`;
